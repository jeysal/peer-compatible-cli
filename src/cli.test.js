// @flow

import execa from 'execa';

test('finds @angular/material versions for angular 2.0.0', async () => {
  const ngMaterialVersions = await execa.stdout('babel-node', [
    'src/cli.js',
    '@angular/material',
    '@angular/core@2.0.0',
  ]);
  expect(ngMaterialVersions).toMatchSnapshot();
});

test('ignores peers not required at all', async () => {
  const ngMaterialVersions = await execa.stdout('babel-node', [
    'src/cli.js',
    '@angular/material',
    '@angular/core@2.0.0',
    'react@15.0.0',
  ]);
  expect(ngMaterialVersions).toMatchSnapshot();
});
