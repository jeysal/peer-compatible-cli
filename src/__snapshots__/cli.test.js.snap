// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finds @angular/material versions for angular 2.0.0 1`] = `
"2.0.0-alpha.9-experimental
2.0.0-alpha.9-experimental-pizza
2.0.0-alpha.9-experimental-5
2.0.0-alpha.9
2.0.0-alpha.9-1
2.0.0-alpha.9-2
2.0.0-alpha.9-3
2.0.0-alpha.10-experimental-yogurt
2.0.0-alpha.10-experimental-coffee
2.0.0-alpha.10-experimental-3
2.0.0-alpha.10-experimental-4
2.0.0-alpha.10-experimental5
2.0.0-alpha.10-experimental6
2.0.0-alpha.10"
`;

exports[`ignores peers not required at all 1`] = `
"2.0.0-alpha.9-experimental
2.0.0-alpha.9-experimental-pizza
2.0.0-alpha.9-experimental-5
2.0.0-alpha.9
2.0.0-alpha.9-1
2.0.0-alpha.9-2
2.0.0-alpha.9-3
2.0.0-alpha.10-experimental-yogurt
2.0.0-alpha.10-experimental-coffee
2.0.0-alpha.10-experimental-3
2.0.0-alpha.10-experimental-4
2.0.0-alpha.10-experimental5
2.0.0-alpha.10-experimental6
2.0.0-alpha.10"
`;
